### PR TITLE
Reject eth_sendTransaction jsonrpc call with error

### PR DESCRIPTION
**Pull request by ZeljkoBenovic**
# Description

Fixes EDGE-17

This PR implements error response on jsonrpc call to eth_sendTransaction method.
We're rejecting this call, as we do not support wallet management
Response example:
```
{"jsonrpc":"2.0","id":1,"error":{"code":-32600,"message":"wallet management is not supported, any request call to the eth_sendTransaction method is not supported"}}
```

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

File: e2e/websocket_test.go [hunk 1]
```diff
@@ -117,6 +118,8 @@ func TestWS_Response(t *testing.T) {
 		assert.Equalf(t, 0, preminedAccounts[0].balance.Cmp(foundBalance), "Balances don't match")
 	})
 
+	// skiping test as we disabled eth_sendTransaction jsonrpc call
+	t.SkipNow()
```
Comment (zivkovicmilos) [inline e2e/websocket_test.go]: Why not put this test skip at the beginning of the test?

File: jsonrpc/eth_endpoint.go [hunk 2]
```diff
@@ -147,18 +148,11 @@ func (e *Eth) SendRawTransaction(input string) (interface{}, error) {
...
 func (e *Eth) SendTransaction(arg *txnArgs) (interface{}, error) {
-	transaction, err := e.d.decodeTxn(arg)
-	if err != nil {
-		return nil, err
-	}
-
-	if err := e.d.store.AddTx(transaction); err != nil {
-		return nil, err
-	}
+	errMsg := "wallet management is not supported, any request call to the eth_sendTransaction method is not supported"
```
Comment (zivkovicmilos) [inline jsonrpc/eth_endpoint.go]: You can inline this error message with the return

File: jsonrpc/types.go [hunk 3]
```diff
@@ -210,9 +210,9 @@ func (a argBig) MarshalText() ([]byte, error) {
 	return []byte("0x" + b.Text(16)), nil
 }
 
-func argAddrPtr(a types.Address) *types.Address {
-	return &a
-}
+//func argAddrPtr(a types.Address) *types.Address {
```
Comment (zivkovicmilos) [inline jsonrpc/types.go]: Please remove this comment block if it's not needed

**Review by zivkovicmilos — APPROVED** [review]
Looks good 👍 

I've left some minor suggestions 💯

**Review by 0xAleksaOpacic — COMMENTED** [review]
Well Done! I have just one suggestion that we can discuss in sync.
@zivkovicmilos @lazartravica I do not think it is good practice to have t.SkipNow() in tests in develop. I also do not think we should delete all of these tests, we should adjust them so they are left with parts of the code where they test something. And if that other part does not make sense to leave we should delete test.

**Review by 0xAleksaOpacic — APPROVED** [review]

File: e2e/websocket_test.go (hunk 1 above)
Comment (ZeljkoBenovic) [reply e2e/websocket_test.go]: The first part of the test is making sure that the websocket endpoint is working properly. We can remove this test entirely but than we won't have any WS tests at all. Should we remove this test entirely ?

File: jsonrpc/eth_endpoint.go (hunk 2 above)
Comment (ZeljkoBenovic) [reply jsonrpc/eth_endpoint.go]: FIxed in commit cbc116c8211dc3ecd4685616b182a8d6db4cb005

File: jsonrpc/types.go (hunk 3 above)
Comment (ZeljkoBenovic) [reply jsonrpc/types.go]: Fixed in commit cbc116c8211dc3ecd4685616b182a8d6db4cb005

**Review by lazartravica — CHANGES_REQUESTED** [review]
@ZeljkoBenovic take a look at Aleksa's comment.

We don't want to:

1. Skip tests.
2. Have commented out tests

We want to only have the tests that bring us value and that's it, remove what doesn't fit in the picture with the disabled JSON-RPC call and that's it.

**Comment by ZeljkoBenovic** [comment]
Removed all skipped / unused tests with commits c6a7fa65f72e068ac822491f277f0d7bcb53fc01 and 1a8b529253b69888d6846c4dd1b394579c00d212

**Review by zivkovicmilos — CHANGES_REQUESTED** [review]
Please revert the test deletions as we discussed 🙏

File: e2e/websocket_test.go (hunk 1 above)
Comment (ZeljkoBenovic) [reply e2e/websocket_test.go]: Reworked all test in 47c3b97be431be3734c715bbe10123d00d9b5f59

**Comment by ZeljkoBenovic** [comment]
I've restored all deleted files/tests and fixed all methods to use sendRawTransacion instead of sendTransaction. Local testing now shows no issues.

File: e2e/framework/testserver.go [hunk 4]
```diff
@@ -417,11 +418,29 @@ func (t *TestServer) SendRawTx(
 	signer := crypto.NewEIP155Signer(100)
 	client := t.JSONRPC()
 
+	// if there is no from addr passed in we can find it in test server configuration
+	var blankAddr types.Address
+	if tx.From == blankAddr {
+		tx.From = t.Config.PremineAccts[0].Addr
+	}
```
Comment (zivkovicmilos) [inline e2e/framework/testserver.go]: Why do we need this?

File: e2e/framework/testserver.go [hunk 5]
```diff
@@ -417,11 +418,29 @@ func (t *TestServer) SendRawTx(
...
+		tx.GasPrice = big.NewInt(DefaultGasPrice)
+	}
+
+	if tx.Gas == 0 {
+		tx.Gas = DefaultGasLimit
+	}
+
+	if tx.Value == big.NewInt(0) || tx.Value == nil {
+		tx.Value = big.NewInt(0)
+	}
```
Comment (zivkovicmilos) [inline e2e/framework/testserver.go]: Why the default values?

I don't think we should override values like this in a common func such as `SendRawTx`

File: e2e/framework/testserver.go [hunk 6]
```diff
@@ -490,12 +509,16 @@ func (t *TestServer) WaitForReady(ctx context.Context) error {
 	return err
 }
 
-func (t *TestServer) TxnTo(ctx context.Context, address web3.Address, method string) *web3.Receipt {
+//nolint:lll
```
Comment (zivkovicmilos) [inline e2e/framework/testserver.go]: Remove the `nolint`, just break up the method declaration on multiple lines so it's readable

**Comment by zivkovicmilos** [comment]
Please check out the changes I've made in https://github.com/0xPolygon/polygon-edge/pull/370.

This PR will be merged out soon, so you can use the fixed up common TestServer methods in this PR

File: e2e/framework/testserver.go (hunk 4 above)
Comment (ZeljkoBenovic) [reply e2e/framework/testserver.go]: Some tests were failing with not enough funds error message. 
This makes sure that that the sender always has enough funds, even if calling function doesn't specify From field. 

Should I investigate exactly which tests are failing when this check is omitted ?

File: e2e/framework/testserver.go (hunk 5 above)
Comment (ZeljkoBenovic) [reply e2e/framework/testserver.go]: tx.Value needs to be set, as SignTx function is panicking when it tries to do tx.Copy and the tx.Value is not set. [here](https://github.com/0xPolygon/polygon-edge/blob/48ec5c46a89b5e57fc538eac9d8e57ebdb350286/types/transaction.go#L53)
 
As for tx.Gas and tx.GasPrice , these same checks are done in SendTxn function ( which we're using right now ) , and I've just implemented them here. Should we remove this ?

File: jsonrpc/eth_endpoint.go [hunk 7]
```diff
@@ -210,18 +210,10 @@ func (e *Eth) SendRawTransaction(input string) (interface{}, error) {
...
-		return nil, err
-	}
-
-	if err := e.store.AddTx(transaction); err != nil {
-		return nil, err
-	}
-
-	return transaction.Hash.String(), nil
+	//nolint: lll
+	return nil, fmt.Errorf("request calls to eth_sendTransaction method are not supported, use eth_sendRawTransaction insead")
```
Comment (zivkovicmilos) [inline jsonrpc/eth_endpoint.go]: This should be broken up into multiple lines and this error extracted into a standardized error with `errors.New`

File: e2e/framework/testserver.go (hunk 4 above)
Comment (zivkovicmilos) [reply e2e/framework/testserver.go]: What tests fail when you omit this?

**Review by zivkovicmilos — APPROVED** [review]
Looks good, please resolve the leftover convos and we're good to go 🚀

File: e2e/framework/testserver.go (hunk 5 above)
Comment (zivkovicmilos) [reply e2e/framework/testserver.go]: Resolving because it's outdated

File: e2e/framework/testserver.go (hunk 4 above)
Comment (ZeljkoBenovic) [reply e2e/framework/testserver.go]: I can't reproduce this fail anymore. Removed in 6138b95b4526bbbefa2f06a54581e344e2ebc9cd

File: e2e/framework/testserver.go (hunk 6 above)
Comment (ZeljkoBenovic) [reply e2e/framework/testserver.go]: Fixed in 6138b95b4526bbbefa2f06a54581e344e2ebc9cd

**Review by Kourin1996 — APPROVED** [review]

**Review by lazartravica — APPROVED** [review]
Confirmed it works.

Thanks!

**Pull request merged**